lib/snitcher.rb

```diff
@@ -29,6 +29,17 @@ module Snitcher
   #
   # @return [Boolean] if the check-in succeeded.
   def snitch!(token, opts = {})
+    if block_given?
+      snitch_start = Time.now
+      begin
+        result = yield
```

Comment:
No point in storing the result if we don't use it. Documentation says we return a boolean but code looks like it returns the HTTP response.

I'm wondering if we should return the result of the block and if we should raise any exceptions (after checking in of course).

```suggestion
        yield
```

Reply:
I started tweaking this, I'll push where I got before lunch… I realized that rescuing the block will cause issues with how `snitch!` vs ~snitch` work. I think I have a plan for a workaround.